src/CMakeLists.txt

```diff
@@ -265,34 +273,64 @@ add_library(${src_lib_static_nupiccore_solo} STATIC
     ${types_files} 
     ${utils_files} 
 )
-
-
-target_compile_options( ${src_lib_static_nupiccore_solo} PUBLIC ${INTERNAL_CXX_FLAGS})
-target_compile_definitions(${src_lib_static_nupiccore_solo} PRIVATE ${COMMON_COMPILER_DEFINITIONS})
-target_include_directories(${src_lib_static_nupiccore_solo} PRIVATE 
+# shared libraries need PIC
+target_compile_options( ${src_objlib} PUBLIC ${INTERNAL_CXX_FLAGS})
+set_property(TARGET ${src_objlib} PROPERTY POSITION_INDEPENDENT_CODE 1)
```

Comment:
I think -fPIC is set in /CommonCompiler.cmake, so rm it there

Reply:
I think it best to remove it from here because ALL compiles must be -fPIC, even the dependencies, so that they can be included in the Python extension.

Reply:
Ok, but better use this version 
> set_property(TARGET ${src_objlib} PROPERTY POSITION_INDEPENDENT_CODE 1) 

then the current there, as this will be more portable 

> -fPIC

Reply:
On windows it is /MD or /MDd  (which is not really "position independent code")
On everything else the compile option is -fPIC  
Both are set in CommonCompilerConfig when building the compiler flags.
Don't think I can use the POSITION_INDEPENDENT_CODE property there.